libcaf_net/caf/policy/scribe.hpp

```diff
@@ -0,0 +1,160 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2018 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#pragma once
+
+#include <caf/error.hpp>
+#include <caf/fwd.hpp>
+#include <caf/logger.hpp>
+#include <caf/net/endpoint_manager.hpp>
+#include <caf/net/receive_policy.hpp>
+#include <caf/net/stream_socket.hpp>
+#include <caf/sec.hpp>
+#include <caf/span.hpp>
+#include <caf/variant.hpp>
+
+#ifdef CAF_WINDOWS
+#  include <winsock2.h>
+#else
+#  include <caf/actor.hpp>
+#  include <sys/socket.h>
+#  include <sys/types.h>
+#endif
+
+namespace caf {
+namespace policy {
+
+class scribe {
+public:
+  explicit scribe(net::stream_socket handle);
+
+  net::stream_socket handle_;
```

Comment:
Why should the constructor be moved to the private block? There would be no way to construct scribes since there is no such thing as a `scribe_factory`

Reply:
Not the constructor, the line `net::stream_socket handle_;`